dask_kubernetes/operator/kubecluster/kubecluster.py

```diff
@@ -339,11 +339,16 @@ async def _create_cluster(self):
                     body=data,
                 )
             except kubernetes.client.ApiException as e:
-                raise RuntimeError(
-                    "Failed to create DaskCluster resource. "
-                    "Are the Dask Custom Resource Definitions installed? "
-                    "https://kubernetes.dask.org/en/latest/operator.html#installing-the-operator"
-                ) from e
+                if e.status == 422:
+                    # unsure how to extract the value of the invalid kwarg
+                    # from the API response to populate below
+                    raise ValueError("unable to process invalid kwarg(s)")
+                elif e.status == 404:
+                    raise RuntimeError(
+                        "Failed to create DaskCluster resource. "
+                        "Are the Dask Custom Resource Definitions installed? "
+                        "https://kubernetes.dask.org/en/latest/operator.html#installing-the-operator"
+                    ) from e
```

Comment:
This is interesting. Does it always give a 404 if the CRDs are missing? And is that the only situation that it would raise a 404?

If so I would just simplify this right down and let the Kubernetes error bubble up to the user.

```
if e.status == 404:
    raise RuntimeError(...) from e
else:
    raise e
```

Reply:
Also I am confused, in what case would the CRD be missing? 

**IF**  the code automatically generate a custom resource  spec with default values via **make_cluster_spec()**  if one isn't passed to `KubeCluster()` ..

then creates the CRD via **create_namespaced_custom_object()**  ?

Reply:
I mean the case where the user has skipped installing the operator and just tries to use `KubeCluster`. When I say CRD I mean Custom Resource **Definition** which describes the custom resource. These get installed by the helm chart. Pytest also installs/removes these via a fixture for testing.

What you are referring to that `create_namespaced_custom_object` creates is an instance of that CRD, usually just referred to as a *resource*. Without the CRDs Kubernetes has no idea what a `DaskCluster` or `DaskWorkerGroup` even is, and it looks like it returns a 404 in this case.

So my guess is that it returns a 404 if the CRDs are missing, and a 422 if you try and create something that doesn't meet the CRDs expectations.